hooks/pre-command

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+set -ueo pipefail
+
+DIR="$(cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd)"
+
+# shellcheck source=lib/shared.bash
+. "$DIR/../lib/shared.bash"
+
+builder_create="$(plugin_read_config BUILDER_CREATE "false")"
+builder_use="$(plugin_read_config BUILDER_USE "false")"
+builder_name="$(plugin_read_config BUILDER_NAME "")"
+
+if [[ "${builder_create}" == "true" ]] || [[ "${builder_use}" == "true" ]]; then
+    if [[ -z "${builder_name}" ]]; then
+        echo "+++ 🚨 Builder Name cannot be empty when using 'create' or 'use' parameters"
+        exit 1
+    fi
+    # exporting to use in pre-exit
+    export builder_name
```

Comment:
why is this necessary? couldn't you just use `$(plugin_read_config BUILDER_NAME "")` in the pre-exit hook as well?

Reply:
trying to be DRY but yeah makes it easier and clearer to have the `pre-exit` hook do environent variable reading